crates/engraver/tests/it/accidentals.rs

```diff
@@ -37,3 +39,10 @@ fn accidental_when_alteration_repeated_over_barline() {
 fn accidental_canceled_in_next_bar() {
     assert_staff_snapshot!("f# | f |");
 }
+
+#[test]
+fn key_signature_alteration() -> Result<()> {
+    assert_staff_snapshot!("c d# e f | c d# e# f ||");
+
+    Ok(())
+}
```

Comment:
I think this could be tested by making the existing `accidental_when_alteration_repeated_over_barline` more thorough:

```rs
#[test]
fn accidental_when_alteration_repeated_over_barline() {
    assert_staff_snapshot!("### f# a# b | f# a# b |");
}
```

(I also wanted to make sure to try it with an actual key signature in case there was another bug there.)